### ISSUE TITLE
predict function not predicting pixels with last level of categorical predictor

### ISSUE THREAD

**Issue by ezlonis** [rspatial/raster#123]
Hello,

I'm using the predict function to predict a randomForest (classification - 2 classes) object on a raster stack.  There are 11 predictor variables, one of which is categorical.  The predict function runs as usual with no warnings or errors, but when inspecting the resultant raster, there are blank pixels in all of the locations for the last two levels of the categorical predictor (14 total levels - any pixels that have levels [raster Value] 13 and 14 are not predicted).  

I reduced the number of levels in the the categorical predictor from 14 to 11 by reclassification and re-ran, then only the last level in not predicted.  I reduced from 11 to 9 levels and still only the last level is not predicted.  I switched levels around so the second to last level was then the last and vice-versa - still the last level is not predicted.  All other pixels have predictions.    

Any ideas what my issue is?

#Random Forest Model
TopModel_CoverReclass3_Test <- randomForest(y=qrdata[,4], x=qrdata[,c(5:15)], ntree = 1000)
    
    # CREATE LIST OF RASTERS
    rlist_CoverReclass3=list.files(getwd(), pattern="tif$", full.names=TRUE) 
    # CREATE RASTER STACK
    xvars_CoverReclass3 <- stack(rlist_CoverReclass3)
    
    #predict
    predict(xvars_CoverReclass3, TopModel_CoverReclass3_Test, filename="CavityPred_CoverReclass3_Test.img", type="response", 
            index=1, na.rm=TRUE, progress="window", overwrite=TRUE)

**Comment by rhijmans** [comment]
Can you provide a reproducible example? If you cannot do it here, feel free to send me an email.

**Comment by ezlonis** [comment]
Hello Dr.Hijmans,

Thank you for your quick reply and offer to solve the issue.  I think I partially solved the mystery.  It seems that there were no training data in the randomForest model for 2 of the 14 levels of the categorical predictor and instead of not making predictions on those two levels (level 5 and 12), predict did not make predictions on the last 2 levels 13 and 14 (which did have training data).  When I reclassified, I must have eliminated one of the levels without training data (by recombining with a level that did have training data) and thus only one (still the highest Value level, which DID have training data) was not predicted.  I further reclassified down to 9 levels and the problem persisted - now the level without training data was level 8 (with 8 being predicted on and 9 not being predicted, despite 9 having training data).  I flipped the classification of 8 and 9 and still 9 was not predicted (9 was then the one without training data).  I reclassified again, making level 9 NoData, and then all 8 levels had training data and were all predicted on...  

Not sure if you would call this user error, a bug, or both :).  let me know if you still want data and code to reproduce the issue and I can put them together for you.   

Thanks!
